src/test/transaction_tests.cpp

```diff
@@ -367,6 +366,15 @@ BOOST_AUTO_TEST_CASE(tx_invalid)
     }
 }
 
+BOOST_AUTO_TEST_CASE(tx_no_inputs)
+{
+    TxValidationState state;
+    bool isValid = CheckTransaction(CTransaction(CMutableTransaction()), state);
+
+    BOOST_CHECK_MESSAGE(!isValid, "Transaction with no inputs should be invalid.");
+    BOOST_CHECK(state.GetRejectReason() == "bad-txns-vin-empty");
```

Comment:
To check that unexpected reject reasons would cause test failure, added characters to the end of the reject reason check.  Test failed as expected.

```c
BOOST_CHECK(state.GetRejectReason() == "bad-txns-vin-empty_ShouldCauseTestFailure");
```

Reply:
Re-ran this sanity check on e2e8b21240da8d567a37d35ae29a6100fde65d2b
Unit test failed as expected.